### ISSUE TITLE
graph traversal: specifying an edge collection twice makes query super slow

### ISSUE THREAD

**Issue by thwd-duda** [arangodb/arangodb#1894]
I executed a query like the following:

```
FOR v IN 1..10 ANY "a/b" ANY a_a, ANY a_a // note edge collection specified twice
    RETURN v
```

Where `a_a` is an edge collection containing relations between objects in the document collection `a` and other objects in that same collection. There's a total of 12 documents in collection `a` and 20 edges in `a_a`.

In my application, the cursor bound to this query timed out. Executing the same query on aardvark yielded a result after 19.590s.

The result was equivalent to just executing the following:

```
FOR v IN 1..10 ANY "a/b" ANY a_a
    RETURN v
```

(The above query executed in 0.043s.)

```
$ uname -a
Linux [redacted] 4.4.0-24-generic #43-Ubuntu SMP Wed Jun 8 19:27:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ arangod --version
architecture: 64bit
build-date: 2016-05-14 15:31:11
configure:  './configure' 'CC=gcc' 'CXX=g++' 'CFLAGS=-g -O3 -fomit-frame-pointer' 'CXXFLAGS=-g -O3 -fomit-frame-pointer' '--enable-gold' '--enable-tcmalloc' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib' '--libexecdir=/usr/lib' '--localstatedir=/var' '--sharedstatedir=/usr/com' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--enable-all-in-one-etcd' '--enable-internal-go'
env:  CFLAGS='-g -O3 -fomit-frame-pointer' CXXFLAGS='-g -O3 -fomit-frame-pointer' CC='gcc' CXX='g++'
fd-client-event-handler: poll
fd-setsize: 1024
icu-version: 54.1
libev-version: 4.11
maintainer-mode: false
openssl-version: OpenSSL 1.0.2g-fips  1 Mar 2016
server-version: 2.8.9
sizeof int: 4
sizeof void*: 8
tcmalloc: true
v8-version: 4.3.61
```

**Comment by jsteemann** [comment]
The direction `ANY` can easily lead to combinatoric explosion.
For example, with just 4 vertices and 5 edges, my traversal produces 25 edges:
```
arangod> db.v.count();
4
arangod> db.e.count();
5
arangod> q = "FOR v IN 1..10 ANY 'v/test1' e RETURN v"; db._query(q).toArray().length;
25
```
Specifying the edge collection repeatedly produces much results, so the result size can quickly get out of hand:
```
arangod> q = "FOR v IN 1..10 ANY 'v/test1' e, ANY e RETURN v"; db._query(q).toArray().length;
222
arangod> q = "FOR v IN 1..10 ANY 'v/test1' e, ANY e, ANY e RETURN v"; db._query(q).toArray().length;
927
arangod> q = "FOR v IN 1..10 ANY 'v/test1' e, ANY e, ANY e, ANY e RETURN v"; db._query(q).toArray().length;
2668
arangod> q = "FOR v IN 1..10 ANY 'v/test1' e, ANY e, ANY e, ANY e, ANY e RETURN v"; db._query(q).toArray().length;
6165
```
However, I think collections should be made unique for the traversals, which will fix the problem when the same collection is specified multiple times.

**Comment by jsteemann** [comment]
Edge collections names are now made unique before the traversal is started. This fixes issues if a collection is specified multiple times. Fixed in 3.0.1 and devel.
